uid/snowflake.go

```diff
@@ -29,6 +29,10 @@ func New() ID {
 }
 
 func (u ID) String() string {
+	if u == 0 {
```

Comment:
This should be superseded by #1860 which fixes a number of snowflake related issues